client/src/components/ViewPostDialog.tsx

```diff
@@ -139,19 +139,22 @@ function MoreOptions(props: {
 
 /* Like button. Handles liking/unliking a post */
 function LikeButton(props: {
+  interactionBit: boolean;
+  setInteractionBit: React.Dispatch<React.SetStateAction<boolean>>;
   numLikes: number;
   doesUserLike: string;
   id: string;
 }) {
   let numLikes = !isNaN(props.numLikes) ? props.numLikes : 0;
-  const isLiked = props.doesUserLike == '1';
+  const isLiked = props.doesUserLike === '1';
 
   const handleClick = async () => {
     if (!isLiked) {
       await api.likePost(props.id);
     } else {
       await api.unlikePost(props.id);
     }
+    props.setInteractionBit(!props.interactionBit);
```

Comment:
If you just want to set it to the opposite value ,you can just do `props.setInteractionBit((prev) => !prev)`.. you dont need `props.interactionBit`